src/pages/authentication/authSlice.js

```diff
@@ -0,0 +1,99 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import {
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+  signOut,
+} from 'firebase/auth';
+import { collection, doc, setDoc } from 'firebase/firestore';
+import { auth, db } from '../../firebase';
+
+const initialState = {
+  currentUser: null,
+  status: 'idle',
+};
+
+export const signUpUser = createAsyncThunk(
+  'auth/signUpUser',
+  async ({ email, password, name, username }, thunkAPI) => {
+    try {
+      const response = await createUserWithEmailAndPassword(
+        auth,
```

Comment:
* Suggestion : You can keep the function name simple and short like createUser, it would be self explanatory

Reply:
@JowelTisso thats  a inbuilt function in firebase